### DISCUSSION TITLE
User permissions: Automatically select all collections resulting from a search

### DISCUSSION THREAD

**Discussion by AlexisPiST** [dani-garcia/vaultwarden#3564] [Q&A]
Hello,

As the group functionality doesn't yet exist, I was wondering if it was possible, when editing a user, to select all the collections resulting from a search in the "Select collections" field via the web GUI.

This would make it possible to quickly add all the child or resultant collections of a search instead of selecting the collections in the list one by one.

Example:

I edit a user's permissions and enter the term "IT" in the "Select collection" search field, which displays all the IT/* collections.
With this in mind, I'm looking for a way to select all the searches and therefore all the child collections of my IT search.

It's not possible to do this in SQL because the names of the collections are encrypted.

Any tips or tricks used by other members?

Thank you for your replies!

**Comment by BlackDex** [comment]
> Hello,
> 
> As the group functionality doesn't yet exist, I was wondering if it was possible, when editing a user, to select all the collections resulting from a search in the "Select collections" field via the web GUI.

Groups are available, but only under a feature flag in the settings, because there are some small bugs.

https://github.com/dani-garcia/vaultwarden/blob/adf67a8ee887d21e104adfa8b6521d7971f5a1f1/.env.template#L93-L98


> 
> This would make it possible to quickly add all the child or resultant collections of a search instead of selecting the collections in the list one by one.
> 
> Example:
> 
> I edit a user's permissions and enter the term "IT" in the "Select collection" search field, which displays all the IT/* collections. With this in mind, I'm looking for a way to select all the searches and therefore all the child collections of my IT search.
> 
> It's not possible to do this in SQL because the names of the collections are encrypted.

The names are encrypted, but the `UUID's` arn't. You should be able to see the `UUID's` when you go the the organization, then click on the collection on the left side. There should appear something like this `vault?collectionId=xxxxxxxx-yyyy-zzzz-aaaa-bbbbbbbbbbbb` in the URL field of the browser.

> 
> Any tips or tricks used by other members?
> 
> Thank you for your replies!

In general, I would suggest to not edit the database manually, since that could cause issues if you make a wrong change. So, always backup the database before you start with something like  you are suggesting.

**Reply by AlexisPiST** [reply]
When you say small bugs, do you have the list?

**Reply by BlackDex** [reply]
This is the one we know about: https://github.com/dani-garcia/vaultwarden/issues/3413

But there could be more. But unless people test, we will never know.
